src/main/scala/mesosphere/marathon/core/group/GroupManager.scala

```diff
@@ -124,12 +125,22 @@ trait GroupManager {
     *              one can control, to stop a current deployment and start a new one.
     * @return the deployment plan which will be executed.
     */
-  def updateRoot(
+  final def updateRoot(
     id: PathId,
     fn: RootGroup => RootGroup,
     version: Timestamp = Group.defaultVersion,
     force: Boolean = false,
-    toKill: Map[PathId, Seq[Instance]] = Map.empty): Future[DeploymentPlan]
+    toKill: Map[PathId, Seq[Instance]] = Map.empty): Future[DeploymentPlan] = {
+    updateRootEither[Nothing](id, fn.andThen(Right(_)), version, force, toKill)
```

Comment:
If we use Either with Nothing, maybe it makes more sense to model this using Option?

Reply:
It's done for the sake of backwards-compatibility with the current code. Once all required methods will accept either, it will be removed.